lightning/src/util/async_poll.rs

```diff
@@ -9,35 +9,66 @@
 
 //! Some utilities to make working with the standard library's [`Future`]s easier
 
+use crate::events::ReplayEvent;
 use crate::prelude::*;
 use core::future::Future;
 use core::marker::Unpin;
 use core::pin::Pin;
 use core::task::{Context, Poll};
 
-pub(crate) struct MultiFuturePoller<F: Future<Output = ()> + Unpin>(pub Vec<Option<F>>);
+enum EventFuture<F: Future<Output = Result<(), ReplayEvent>>> {
+	Pending(F),
+	Ready(Result<(), ReplayEvent>),
+}
+
+pub(crate) struct MultiEventFuturePoller<F: Future<Output = Result<(), ReplayEvent>> + Unpin> {
```

Comment:
Could we parameterize the error type? Seems the utility doesn't need to be specific to event processing.